pkg/registry/apis/alerting/notifications/receiver/conversions.go

```diff
@@ -56,9 +63,29 @@ func convertToK8sResource(orgID int64, receiver *ngmodels.Receiver, namespacer r
 		Spec: spec,
 	}
 	r.SetProvenanceStatus(string(receiver.Provenance))
+
+	if access != nil {
+		for _, action := range ngmodels.ReceiverPermissions() {
+			mappedAction, ok := permissionMapper[action]
+			if !ok {
+				return nil, fmt.Errorf("unknown action %v", action)
+			}
+			if can, _ := access.Has(action); can {
+				r.SetAccessControl(mappedAction)
+			}
+		}
+	}
+
 	return r, nil
 }
 
+var permissionMapper = map[ngmodels.ReceiverPermission]string{
+	ngmodels.ReceiverPermissionReadSecret: "canReadSecrets",
+	//ngmodels.ReceiverPermissionAdmin:      "canAdmin", // TODO: Add when resource permissions are implemented.
+	ngmodels.ReceiverPermissionWrite:  "canWrite",
+	ngmodels.ReceiverPermissionDelete: "canDelete",
+}
```

Comment:
WDYT about sligthly simpler approach - we have `PermissionSet[ReceiverPermission]` where `RecevierPermission` is enum, e.g. `write` , `delete` etc. We can just take this permission and add `can` prefix. We can do it in `SetAccessControl` which will create annotation 
```
o.Annotations[fmt.Sprintf("%s%s/can%s%s", InternalPrefix, "access", strings.ToUpper(action[:1]),action[1:])]
```